src/averageAges.js

```diff
@@ -56,8 +69,31 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+
+  const childs = people.filter(person => onlyWithSon
+    ? people.some(mother => mother.name === person.mother) && person.sex === 'm'
+    : people.some(mother => mother.name === person.mother));
```

Comment:
![image](https://user-images.githubusercontent.com/106398574/179058069-5203036e-8df2-41f6-95fc-c243a57a9389.png)

Reply:
But is is under filter :)